src/main/java/com/opencart/steps/MainPageBL.java

```diff
@@ -1,9 +1,66 @@
 package com.opencart.steps;
 
+import com.opencart.containers.ProductContainer;
+import com.opencart.datamodel.ProductModel;
+import com.opencart.pages.SearchResultPage;
+import org.testng.Assert;
+
+import java.util.List;
+import java.util.Map;
+
 public class MainPageBL {
 
     public HeaderPageBL getHeaderPageBL() {
         return new HeaderPageBL();
     }
+
+    /**
+     * Method finds all ProductContainers on the Page and checks
+     * if their Product Cost in specified currency is correct.
+     *
+     * @param currencyName is name of Currency that needs to be checked.
+     */
+
+    public void verifyAllProductsPrices(Map<String, String> currencies, String currencyName) {
```

Comment:
I believe you can get the currency name from the map. Am I wrong?

Reply:
yes) now I see it.